### PR TITLE
Rewrite the Backend.autoFocusFirstInputField to vanilla JS

### PR DESCRIPTION
**Pull request by aschempp**

### CODE REVIEW

**Review by Toflar — DISMISSED** [review]

File: core-bundle/src/Resources/public/core.js [hunk 1]
```diff
@@ -906,15 +906,6 @@ var Backend =
...
 
-	/**
-	 * Toggle the synchronization results
-	 */
-	toggleUnchanged: function() {
-		$$('#result-list .tl_confirm').each(function(el) {
-			el.toggleClass('hidden');
-		});
-	},
-
```
Comment (ausi) [inline core-bundle/src/Resources/public/core.js:822]: Seems to be unrelated? On purpose?

File: core-bundle/src/Resources/public/autofocus.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    const inputs = edit.querySelectorAll('input, textarea');
+
+    for (let i = 0; i < inputs.length; i++) {
+        const input = inputs[i];
+
+        if (
+            !input.getAttribute('disabled')
+            && !input.getAttribute('readonly')
```
Comment (ausi) [inline core-bundle/src/Resources/public/autofocus.js]:
```suggestion
            !input.disabled
            && !input.readonly
```

The mootools `get()` method is not synonymous with `getAttribute`.

File: core-bundle/src/Resources/public/autofocus.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+            && !isHidden(input)
```
Comment (ausi) [inline core-bundle/src/Resources/public/autofocus.js]:
```suggestion
            && input.offsetWidth
            && input.offsetHeight
```

File: core-bundle/src/Resources/public/core.js (hunk 1 above)
Comment (aschempp) [reply core-bundle/src/Resources/public/core.js:822]: nah, I just forgot it in https://github.com/contao/contao/pull/4327 and this was the next PR related to JS 😆

File: core-bundle/src/Resources/public/autofocus.js (hunk 3 above)
Comment (aschempp) [reply core-bundle/src/Resources/public/autofocus.js]: That would not check for the `display: none` property (according to the Mootols code)?

File: core-bundle/src/Resources/public/autofocus.js (hunk 2 above)
Comment (aschempp) [reply core-bundle/src/Resources/public/autofocus.js]: According to the Mootols code, it actually is pretty synonymous. Even though `getAttribute` and DOM properties are _almost always_ synonymous (they are in this case), https://javascript.info/dom-attributes-and-properties suggests to use the DOM property since it would probably also be boolean in this case. I'll adjust all my PRs 👍

Comment (ausi) [reply core-bundle/src/Resources/public/autofocus.js]: Side note: using `getAttribute()` for boolean attributes gives you the empty string for the `true` value, need to use `hasAttribute()` instead then to get a boolean.
But 👍 for using DOM properties instead.

File: core-bundle/src/Resources/public/autofocus.js (hunk 3 above)
Comment (ausi) [reply core-bundle/src/Resources/public/autofocus.js]: Elements with `display: none` have `offsetWidth` and `offsetHeight` of `0`.

File: core-bundle/src/Resources/public/autofocus.js (hunk 2 above)
Comment (aschempp) [reply core-bundle/src/Resources/public/autofocus.js]: Changed in b4b4b733ab915ced966e20ea3f073d28bc27d52f

File: core-bundle/src/Resources/public/autofocus.js (hunk 3 above)
Comment (aschempp) [reply core-bundle/src/Resources/public/autofocus.js]: Changed in b4b4b733ab915ced966e20ea3f073d28bc27d52f

**Review by leofeyer — APPROVED** [review]
